engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheHashingJobActorSpec.scala

```diff
@@ -0,0 +1,292 @@
+      // md5 of 0
+      HashResult(HashKey("output count"), HashValue("CFCD208495D565EF66E7DFF9F98764DA")),
+      HashResult(HashKey("runtime attribute: docker"), HashValue("N/A")),
+      // md5 of 1
+      HashResult(HashKey("runtime attribute: cpu", checkForHitOrMiss = false), HashValue("C4CA4238A0B923820DCC509A6F75849B")),
+      // md5 of 0
+      HashResult(HashKey("runtime attribute: continueOnReturnCode"), HashValue("CFCD208495D565EF66E7DFF9F98764DA")),
+      // md5 of "hello" (with quotes)
+      HashResult(HashKey("input: String stringInput"), HashValue("5DEAEE1C1332199E5B5BC7C5E4F7F0C2"))
+    )
+    val expectedAggregatedInitialHash = "67A8FA54B6D3A57D40AA6FE39C6C3992"
+    val expectedInitialHashResult = InitialHashingResult(expectedInitialHashes, expectedAggregatedInitialHash)
+    parent.expectMsg(expectedInitialHashResult)
+    callCacheRead.expectMsg(expectedInitialHashResult)
+    actorUnderTest.stateName shouldBe WaitingForHashFileRequest
+    actorUnderTest.stateData shouldBe CallCacheHashingJobActorData(
+      List(SingleFileHashRequest(jobDescriptor.key, HashKey("input: File fileInput"), WdlFile("world"), None)),
+      Option(callCacheRead.ref)
+    )
+  }
+  
+  def makeCCHJA(callCacheReader: Option[ActorRef],
+                testFileHashingActor: ActorRef,
+                parent: ActorRef,
+                writeToCache: Boolean = true,
+                addFileHashMockResult: Option[(CallCacheHashingJobActorData, Option[CCHJAFileHashResponse])] = None) = {
+    TestFSMRef(new CallCacheHashingJobActor(
+      templateJobDescriptor(),
+      callCacheReader,
+      None,
+      Set.empty,
+      "backend",
+      Props.empty,
+      writeToCache = writeToCache
+    ) {
+      override def makeFileHashingActor() = testFileHashingActor
+      override def addFileHash(hashResult: HashResult, data: CallCacheHashingJobActorData) = {
+        addFileHashMockResult.getOrElse(super.addFileHash(hashResult, data))
+      }
+    }, parent)
+  }
+  
+  it should "send hash file requests when receiving a NextBatchOfFileHashesRequest" in {
+    val callCacheReadProbe = TestProbe()
+    val fileHashingActor = TestProbe()
+    // self acts as the parent
+    val cchja = makeCCHJA(Option(callCacheReadProbe.ref), fileHashingActor.ref, TestProbe().ref)
+
+    val fileHashRequest1 = mock[SingleFileHashRequest]
+    val fileHashRequest2 = mock[SingleFileHashRequest]
+    cchja.setState(
+      WaitingForHashFileRequest,
+      CallCacheHashingJobActorData(List(List(fileHashRequest1, fileHashRequest2)), List.empty, None)
+    )
+
+    cchja ! NextBatchOfFileHashesRequest
+
+    fileHashingActor.expectMsg(fileHashRequest1)
+    fileHashingActor.expectMsg(fileHashRequest2)
+    cchja.stateName shouldBe HashingFiles
+  }
+
+  it should "send NoFileHashesResult and stop if there are no input files" in {
+    val parent = TestProbe()
+    val callCacheReadProbe = TestProbe()
+    // self acts as the parent
```

Comment:
Note to self: remove those comments they don't apply anymore as we have explicit parent `TestProbe`s now